### ISSUE TITLE
walk.forward analysis: where is test results?

### ISSUE THREAD

**Issue by ntguardian** [braverock/quantstrat#56]
I've tried using the `walk.forward()` function for walk-forward analysis. I get a list reporting the results of the analysis on training sets but never do I see the results for the test set. I've looked in [walk.forward.R](https://github.com/braverock/quantstrat/blob/master/R/walk.forward.R) and testing is done around line 157 of the file, but I don't see it being stored anywhere except maybe in some **blotter** environment, but when I try to look at the portfolio using the method used in the demo, [luxor.8.walk.forward.R](https://github.com/braverock/quantstrat/blob/master/demo/luxor.8.walk.forward.R), which is to use `tradeStats(portfolio.st)`, I get `NULL`. (The demo doesn't work either; I get the error message `Error in runSum(x, n) : Invalid 'n'`, in addition to warning messages. This happens every time. I've had this come up and I think it's related to training or testing periods being smaller than a window size for a moving average being tried.)

Am I missing something or does `walk.forward()` just not save testing results anywhere?

**Comment by braverock** [comment]
The problems in luxor.8.walk.forward are discussed in #34.

All data should be saved if `audit=TRUE` and `include.insamples=TRUE` all data will be stored for both the training and testing periods.

If this is not your experience, please provide a minimal reproducible example.
